pkg/controller/awsfederatedrole/awsfederatedrole_controller.go

```diff
@@ -139,12 +139,19 @@ func (r *ReconcileAWSFederatedRole) Reconcile(request reconcile.Request) (reconc
 		NameSpace:  awsv1alpha1.AccountCrNamespace,
 		AwsRegion:  "us-east-1",
 	})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
 	// Validates Custom IAM Policy
 
 	log.Info("Validating Custom Policies")
 	// Build custom policy in AWS-valid JSON and converts to string
 	jsonPolicy, err := utils.MarshalIAMPolicy(*instance)
+	if err != nil {
+		reqLogger.Error(err, "failed marshalling IAM Policy", "instanceRoleName", instance.Spec.RoleDisplayName)
+		return reconcile.Result{}, err
```

Comment:
ditto... though I'm wondering if this one will get you in a hard spin (if you couldn't marshal it before, what could have changed that you could marshal it the next time?)

Reply:
@2uasimojo: Agreed.  It looks like the only error that would be returned here is one bubbled up from `json.Marshal(&policyDoc)` in the underlying function.  That would indeed just continue to cycle the reconcile loop over and over.

I'm unsure how to handle this, though. It appears that all the errors in this controller just requeue. We could avoid the loop by not returning the error to the reconcile loop, but then the CR updates are just ignored.  I don't believe there are states for this CR that can be marked as "failed" or what not.

Removing the return doesn't change anything, as this will just fail later at https://github.com/openshift/aws-account-operator/blob/0c9e7b4cb0d304012f2d28b61f68666700c0a8d9/pkg/controller/awsfederatedrole/awsfederatedrole_controller.go#L177, and be reconciled with an error, thus looping again.

Reply:
Yeah, I wouldn't worry about it for this PR. If you truly wanted to punt, you could remove the `return` and leave a comment similar to :arrow_up:, but meh.